### ISSUE TITLE
AI spamming certain buildings

### ISSUE THREAD

**Issue by Zervox** [jsettlers/settlers-remake#270]
I've seen that the AI leaves very little open spaces, eg cramping as much as it can into as small area as possible, this is quite counter intuitive for forester-wood works areas or even potentially some were placed ontop of wine growery grow space.
I also notice they build ALOT of bakeries even though they might only have 2-3 grain farms, it seems like it does a 1 grain to 4 bakery ratio, also does seem like a 1 baker 2 water works. 

it also tends to build ALOT of wine growers, and tons of small temples(seems to build one small temple per growery or so, it ends up with alot of both) I've only seen one AI build a gold smeltery, I tend to build more. especially now as the storage space doesn't work yet.(build lots of gold smeltery and destroy them when they've made max amount of bars).

but all in all I loved fighting against the agressive AI.

**Comment by codingberlin** [comment]
Thank you for your feedback!

I structure your feedback in the following issues:
- [ ] optimize space for forester and lumber jacks and eventually their position
- [x] buildings were placed on top of farmers and winegrowers fields
- [ ] optimize ratio between food production buildings
- [ ] optimize number of winegrowers and small temples (best between fast level3 and low material costs for the buildings)
- [ ] increase combat strength by gold

Regarding the farmers and winegrowers fields I checked the item on the issue list because end of last year I fixed that already. The buildings were placed as near as possible on the fields but not on the fields itself. For the human eye it seems so that the border of the fields were blocked by other buildings but I checked it out: if there are no buildings near by at all the winegrower of farmer would not produce more goods. Thus the AI can use as much as possible to build buildings within the own land.

Regarding the combat strength: This is in the development pipeline. I currently wait for the working storages and/or markets. Then I will implement that the AI will use it to store gold.

I will update the above mentioned Issue list in this github Issue every time I fix one item.

I'm delighted to hear that you love it to fight against the very hard AI. I will make it even stronger in the future (more intelligent movements of their soldiers)

**Comment by andreas-eberle** [comment]
@codingberlin: Are some of these tasks still open? Or is that all finished?

**Comment by codingberlin** [comment]
Yes the above mentions task which aren't checked are still open.
The two tasks `optimize ratio between food production buildings` and `optimize number of winegrowers and small temples (best between fast level3 and low material costs for the buildings)` can be fixed after or maybe while the game is balanced (you mentioned in other issues that the balancing will come later when more is finished and not now)

regarding `increase combat strength by gold` I can remember that I tried out to store gold by markets but on the one hand -when my memory is correct- the gold at the marked was not counted and on the other hand in comparisin to the original settlers the remake the bearers take the nearest gold to the market so an endless storage via markeds did not work. So the AI needs to use storages which are not finished yet, are they?

The task 'optimize space for forester and lumber jacks and eventually their position' I didn't start because the wood supply is currently not perfect but good enought so that I focused on other priorities like the pioneers because gather land with pioneers had a bigger economical impact than better wood supply.

**Comment by andreas-eberle** [comment]
Thanks for this update. Indeed, the storage building is not finished yet.
